### PR TITLE
Support external ID's when POSTing users and PUTting groups.

### LINKED ISSUE #2186: API: External ID for User entity cannot be set via API
### Description of the problem

When using `data_source=1` (i.e. configuration data external), User entity cannot be create via API endpoint `/api/v4/users` due to missing externalid attribute

### Your environment
DOMjudge 8.2.1

### Steps to reproduce
- set data_source to 1 (configuration data external)
- make API call
`curl -XPOST https://domserver/api/v4/users -Fusername=demo -Fname=demo -Fpassword= -F"roles[]=team"`

### Expected behaviour
External id can be set via API call or is automatically created. Like in the Teams entity, `id` attribute should be included in the API call:
`curl -XPOST https://domserver/api/v4/users -Fid=demo -Fusername=demo -Fname=demo -Fpassword= -F"roles[]=team"` and that should set the external id if using configuration data external mode.
```
{"last_login_time":null,"last_api_login_time":null,"first_login_time":null,"team":null,"team_id":null,"roles":["team"],"type":"team","id":demo,"username":"demo","name":"demo","email":null,"last_ip":null,"ip":null,"enabled":true}
```

### Actual behaviour
API call returns:
```
{"last_login_time":null,"last_api_login_time":null,"first_login_time":null,"team":null,"team_id":null,"roles":["team"],"type":"team","id":null,"username":"demo","name":"demo","email":null,"last_ip":null,"ip":null,"enabled":true}
```
No external id for the entity is created and the configuration checker shows an error "User XY does not have an external ID" which needs to be fixed by manually editing all User entities via GUI.

### LINKED ISSUE #2381: API: External ID for Group entity cannot be set via API
### Description of the problem

Similar problem to #2186; when using `data_source=1` (i.e. configuration data external), Group entity (a.k.a. Team Categories) is created without externalid when using api endpoint `/api/v4/contests/{cid}/groups`.

### Your environment
DOMjudge 8.3.0DEV/30f885875

### Steps to reproduce
- set data_source to 1 (configuration data external)
- make API call
`curl -XPOST https://domserver/api/v4/contests/demo/groups -Fname=demo -Ficpc_id=1234`

### Expected behaviour
External id can be set via API call or is automatically created. Like in the Teams entity, `id` attribute could be included in the API call:
`curl -XPOST https://domserver/api/v4/contests/demo/groups -Fid=1234 -Fname=demo -Ficpc_id=1234` and that should set the external id if using configuration data external mode.
```
{"hidden":false,"id":1234,"icpc_id":"1234","name":"test","sortorder":0,"color":null,"allow_self_registration":false}
```

### Actual behaviour
API call returns:
```
{"hidden":false,"id":null,"icpc_id":"1234","name":"test","sortorder":0,"color":null,"allow_self_registration":false}
```
No external id for the entity is created and the configuration checker shows an error "Team category XY does not have an external ID" which needs to be fixed by manually editing all entities via GUI.

### PR DESCRIPTION
**Pull request by nickygerritsen**
Fixes #2186.
Fixes #2381.

### CODE REVIEW

File: webapp/src/Controller/API/GroupController.php [hunk 1]
```diff
@@ -112,6 +119,7 @@ public function addAction(
         $idField = $this->eventLogService->externalIdFieldForEntity(TeamCategory::class) ?? 'categoryid';
         $method = sprintf('get%s', ucfirst($idField));
         $id = call_user_func([$group, $method]);
+        dump($group);
```
Comment (vmcj) [inline webapp/src/Controller/API/GroupController.php:122]:
```suggestion
```

File: webapp/src/Controller/API/UserController.php [hunk 2]
```diff
@@ -307,12 +307,17 @@ public function addAction(
         AddUser $addUser,
         Request $request
     ): Response {
+        if ($this->eventLogService->externalIdFieldForEntity(User::class) && !$addUser->id) {
```
Comment (vmcj) [inline webapp/src/Controller/API/UserController.php:310]: Although User is not in CLICS, shouldn't we make this consistent?

File: webapp/src/DataTransferObject/TeamCategoryPost.php [hunk 3]
```diff
@@ -8,6 +8,8 @@
 class TeamCategoryPost
 {
     public function __construct(
+        #[OA\Property(description: 'The ID of the group. Only allowed with PUT requests', nullable: true)]
```
[the remark this answers is not in the document]
Comment (nickygerritsen) [reply webapp/src/DataTransferObject/TeamCategoryPost.php:11]: I have a better idea, hold on.

File: webapp/src/Controller/API/UserController.php (hunk 2 above)
Comment (nickygerritsen) [reply webapp/src/Controller/API/UserController.php:310]: Yeah I was wondering the same.

Comment (nickygerritsen) [reply webapp/src/Controller/API/UserController.php]: Updated to do this.

File: webapp/src/DataTransferObject/TeamCategoryPost.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickygerritsen) [reply webapp/src/DataTransferObject/TeamCategoryPost.php]: I implemented this.

**Review by vmcj — APPROVED** [review]

**Review by meisterT — APPROVED** [review]

**Pull request merged**